src/main/java/org/owasp/esapi/configuration/EsapiPropertyLoader.java

```diff
@@ -33,6 +33,13 @@ public interface EsapiPropertyLoader {
      */
     public Boolean getBooleanProp(String propertyName) throws ConfigurationException;
 
+    /**
+     * Get any Boolean type property from security configuration.
+     * If property does not exist in configuration or has incorrect type, defaultValue is returned
```

Comment:
From the Javadoc: for this interface: "or has incorrect type". 

Well, that is the $64k question, isn't it? What is the _INTUITIVE_ behavior if one calls something like:
```java
    Boolean propValue = getBoolean("SomePropName", true)
```
but somewhere, say in **ESAPI.properties** or elsewhere, we have defined
```bash
SomePropName=This is not a boolean property
```
Then what is the expected / intuitive behavior for that call? I think that I'd expect a `ConfigurationException` to be thrown rather than having `propValue` silently being set to `true`. Or maybe I shouldn't say that is the _intuitive_ behavior, but rather the _**safe**_ behavior. I think because copy/paste errors abound and it would be easy to make a typo with the property name, that's the only way that we're going to help people catch these bizarre accidental edge cases. Otherwise, we're going to cause them to curse at us because they'll have to figure up a debugger to figure out what the hell is going on in cases like this.

@xeno6696 and @jeremiahjstacey  - I'd like your opinion here. I know this requires yet another code change, but if we do a release and document it this way this (likely bug, IMO) will become a feature because changing it later on would break backward compatibility. If we were requiring JDK 9 or later we could just not export this method and keep it hidden, but since Java 8 is the minimal version, once we add this, it becomes available to the world.

Reply:
Overall, I think this is more scope than we should put time into to modify the opt-out request from the original issue.  I appreciate the initiative; however, we're adding this case to expand for something that could have been solved in two copies of an 6 line block.

This is overengineering in this scope.  We're blocking value, and a new possible release for this API change.

My suggestion is to take it out.  Use the duplicate snippet from my previous comment in the 2 factories and revert the changes in the DefaultSecurityConfiguration.  Create another task for the default API that we can address (reuse this code if you want), and complete it in a non-blocking scope.  We don't need this to fix the original problem, and as you can see the discussion of edge cases and caveats will continue to grow.  Let's get the opt-out behavior in in a first form, then improve upon it in another task to expand the SecurityConfiguration API to allow defaults to be specified.

Reply:
I concur. If in the future, you want to do a new PR to get a new interface for
```java
public Boolean getBooleanProp(String propertyName, Boolean defaultValue) throws ConfigurationException
```
I am fine with that, but as my review of this as surfaced, this is a little more nuanced than intuition alone might have us believe. If we are going to add that a future time, I think we need to start out with a GitHub issue and discuss the specifications there first, before waiting for an implementation to discuss them. So, to proceed with this PR, please rip that part of the code out and follow Jeremiah's suggestion.